lib/net/ntlm/encode_util.rb

```diff
@@ -39,7 +39,7 @@ def self.decode_utf16le(str)
       #   the function will convert the string bytes to UTF-16LE and note the encoding as UTF-8 so that byte
       #   concatination works seamlessly.
       def self.encode_utf16le(str)
-        str.dup.force_encoding('UTF-8').encode(Encoding::UTF_16LE, Encoding::UTF_8).force_encoding('UTF-8')
+        str.dup.force_encoding('UTF-8').encode(Encoding::UTF_16LE, Encoding::UTF_8).force_encoding('ASCII-8BIT')
```

Comment:
i think this change makes sense and behaves more like the comment block describes, since its supposed to be safe to perform byte concatenation on the result